idaes/core/solvers/petsc.py

```diff
@@ -279,6 +279,14 @@ def _set_dae_suffixes_from_variables(m, variables, deriv_diff_map):
             dae_var_link_index += 1
             if not diffvar.fixed:
                 differential_vars.append(diffvar)
+            else:
+                raise RuntimeError(
+                    f"Problem cannot contain a fixed differential variable and "
+                    f"unfixed derivative. Consider either fixing the "
+                    f"corresponding derivative or adding a constraint for the "
+                    f"differential variable {diffvar} possibly using an "
+                    f"explicit time variable."
+                )
```

Comment:
Will this catch fixed differential variables at initial conditions?

Reply:
No this only gets called after the initial conditions, so it should be ok.  Tests have fixed differential variables as i.c.s, so we should be good.

Reply:
Okay, sounds good. I'll run this locally then this should be good.